src/vcpkg/sourceparagraph.cpp

```diff
@@ -486,14 +506,104 @@ namespace vcpkg
             else
             {
                 r.add_generic_error(type_name(), std::move(opt).error());
-                return PlatformExpression::Expr::Empty();
+                return PlatformExpression::Expr();
             }
         }
 
         static const PlatformExprDeserializer instance;
     };
     const PlatformExprDeserializer PlatformExprDeserializer::instance;
 
+    struct DefaultFeatureNameDeserializer : Json::IDeserializer<std::string>
+    {
+        virtual LocalizedString type_name() const override { return msg::format(msgADefaultFeature); }
+        virtual Optional<std::string> visit_string(Json::Reader& r, StringView sv) const override
+        {
+            if (sv == "core")
+            {
+                r.add_generic_error(type_name(), msg::format(msgDefaultFeatureCore));
+                return sv.to_string();
+            }
+
+            if (sv == "default")
+            {
+                r.add_generic_error(type_name(), msg::format(msgDefaultFeatureDefault));
+                return sv.to_string();
+            }
+
+            return Json::FeatureNameDeserializer::instance.visit_string(r, sv);
+        }
+        static const DefaultFeatureNameDeserializer instance;
+    };
+
+    const DefaultFeatureNameDeserializer DefaultFeatureNameDeserializer::instance;
+
+    struct DefaultFeatureDeserializer : Json::IDeserializer<DependencyRequestedFeature>
+    {
+        LocalizedString type_name() const override { return msg::format(msgADefaultFeature); }
+
+        constexpr static StringLiteral NAME = "name";
+        constexpr static StringLiteral PLATFORM = "platform";
+
+        Span<const StringView> valid_fields() const override
+        {
+            static const StringView t[] = {
+                NAME,
+                PLATFORM,
+            };
+            return t;
+        }
+
+        Optional<DependencyRequestedFeature> visit_string(Json::Reader& r, StringView sv) const override
+        {
+            return DefaultFeatureNameDeserializer::instance.visit_string(r, sv).map(
+                [](std::string&& name) { return DependencyRequestedFeature{std::move(name)}; });
```

Comment:
```suggestion
            return DefaultFeatureNameDeserializer::instance.visit_string(r, sv);
```
& make `OptionalStorage(Optional<U>)` use `new (&m_t) T{std::move(*p)};`

Reply:
I really don't want to do that because it will accidentally engage initializer_list ctors :/